packages/server/src/user/user.service.ts

```diff
@@ -84,19 +87,41 @@ export class UserService {
     });
 
     // throw error if user exist
-    if (user_count !== 0) {
+    if (userCount !== 0) {
       throw new Error('User already exist in the database.');
     }
 
-    const pwd_hash = await argon2.hash(data.password);
+    const pwdHash = await bcrypt.hash(data.password, this.SALT_ROUNDS);
 
     return await this.prisma.user.create({
       data: {
         project_id: data.project_id,
         username: data.username,
         email: data.email,
-        password: pwd_hash,
-        role: data.role
+        password: pwdHash
+      }
+    });
+  }
+
+  /**
+   * Find all user records in the database
+   *
+   * @returns List of `User` object, will return empty list if no user found
+   */
+  async findAllUsers(): Promise<User[]> {
+    return await this.prisma.user.findMany();
```

Comment:
NIT: Shouldn't have to await here since `findMany` will return a promise here

Reply:
removed